src/placeos-driver/transport.cr

```diff
@@ -66,14 +66,38 @@ abstract class PlaceOS::Driver::Transport
 
         # Make the request
         client = new_http_client(uri, context)
-        client.exec(method.to_s.upcase, uri.request_target, headers, body)
+        check_http_response_encoding client.exec(method.to_s.upcase, uri.request_target, headers, body)
       {% end %}
     end
 
+    protected def check_http_response_encoding(response)
+      headers = response.headers
+      encoding = headers["Content-Encoding"]?
+      if encoding.in?({"gzip", "deflate"})
+        response.consume_body_io
+        body = response.body
+
+        if !body.blank?
+          body_io = IO::Memory.new(body)
+          body = case encoding
+                when "gzip"
+                  Compress::Gzip::Reader.open(body_io) { |gzip| gzip.gets_to_end }
+                when "deflate"
+                  Compress::Deflate::Reader.open(body_io) { |deflate| deflate.gets_to_end }
```

Comment:
minor, but can you use the short block notation for this?

Reply:
yeah good call